### DISCUSSION TITLE
[Question] Generate a 3D mesh from a boundary mesh?

### DISCUSSION THREAD

**Discussion by loumalouomega** [SCIInstitute/Cleaver2#121] [Q&A]
Greetings,

I have recently found your library. Very interesting and looks impressive. My problem is that I work in FEM for pure mechanics purposes and not in biomechanics, neither other medical applied mechanics with FEM. My question is:

**Can I generate a 3D mesh from a boundary mesh?** (pure triangles) In my particular case, I can guarantee it is watertight (so no like the typical STL file with discontinuous geometries). 

I would like to avoid generating a NRRD file with the Signed Distance Field (which I assume it will work), but it would be too much work and many additional steps for what I would like to do.

Best regards,

Vicente Mataix Ferrándiz

**Comment by jessdtate** [comment]
Thanks for your interest in Cleaver!  

Cleaver is a lattice cleaving algorithm, and therefore operates on a background mesh with an indicator function on it (a distance map).  It needs to start with a volume mesh, which can be an irregular tetrahedral mesh.  There really isn't a way around converting your surface mesh to a distance map for Cleaver.  However, converting to a distance map is fairly easy in another tool of ours, [SCIRun](https://github.com/SCIInstitute/SCIRun).

For meshing existing surface meshes, we often use Tetgen. It is implemented in SCIRun, among other tools (gmesh, FEbiostudio, 3D Slicer, etc).  You can check out those tools for more info

**Reply by loumalouomega** [reply]
Excellent, very nice response. For my personal case, I will need to look for an alternative, because I wanted to avoid the steps you are mentioning.

**Reply by loumalouomega** [reply]
Feel free to close the discussion if needed

**Reply by loumalouomega** [reply]
Thank you very much

**Reply by loumalouomega** [reply]
And happy new year!
